pmd-core/src/main/java/net/sourceforge/pmd/lang/Parser.java

```diff
@@ -1,49 +0,0 @@
-/**
- * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
- */
-
-package net.sourceforge.pmd.lang;
-
-import java.io.Reader;
-
-import net.sourceforge.pmd.lang.ast.ParseException;
-import net.sourceforge.pmd.lang.ast.RootNode;
-
-/**
- * Produces an AST from a source file. Instances of this interface must
- * be stateless (which makes them trivially threadsafe).
- *
- * TODO
- *  - Ideally ParserOptions would be an argument to ::parse
- *  - ::parse would also take some more parameters, eg an error collector
- *  - The reader + filename would be a TextDocument
- *
- * @author Pieter_Van_Raemdonck - Application Engineers NV/SA - www.ae.be
- */
-public interface Parser {
```

Comment:
We could keep a interface at the old package to make it easier to transition:

```java
package net.sourceforge.pmd.lang;

@Deprecated
@DeprecatedUntil700
public interface Parser extends net.sourceforge.pmd.lang.ast.Parser {
}
```

Not sure, if that's worth. Regarding these `@DeprecatedUntil700` - we could move these files into a own module called "pmd-compat6".